src/internal/cluster/zarf.go

```diff
@@ -99,21 +102,111 @@ func (c *Cluster) StripZarfLabelsAndSecretsFromNamespaces() {
 	spinner.Success()
 }
 
+// PackageSecretNeedsWait checks if the the package status or any of the component statuses are being mutated by a webhook and need to be waited on.
+func (c *Cluster) PackageSecretNeedsWait(secretName string) (bool, int, error) {
+	// Get the secret that describes the deployed package
+	packageSecret, err := c.GetSecret(ZarfNamespaceName, secretName)
+	if err != nil {
+		return false, 0, err
+	}
+
+	// Parse the secret
+	var deployedPackage types.DeployedPackage
+	err = json.Unmarshal(packageSecret.Data["data"], &deployedPackage)
+	if err != nil {
+		return false, 0, err
+	}
+
+	// Check if there are any component level statuses that we need to wait for
+	for componentName, hookMap := range deployedPackage.ComponentWebhooks {
```

Comment:
This could be more surgical if it was passed the component we are looking for instead of just checking everything all of the time.

Reply:
Great point. Refactored `PackageSecretNeedsWait` to look for the specific component that is passed in